icu4j/main/classes/core/src/com/ibm/icu/impl/ICUResourceBundle.java

```diff
@@ -1396,16 +1396,20 @@ public ICUResourceBundle load() {
 
             // fallback to locale ID parent
             if(b == null){
+                OpenType newOpenType = openType;
+                if (openType == OpenType.LOCALE_DEFAULT_ROOT && localeName.equals(defaultID)) {
```

Comment:
> localeName.equals(defaultID)

Is there a chance that this test fails for locales that have different string ID, but they are in fact the same locale?
Like `he` / `iw`, or `id` / `in` ?
Or even `FR` <> `fr` or `it_IT` <> `it-IT`?

In other words, are we sure things are normalized / canonicalized somewhere, before we get here?

Reply:
I tried all these things, and they worked fine.  The code that deals with obsolete language tags is incorporated into the code that normalizes the locale IDs for things like case and punctuation, so they weren't affected by the problem.  I could add unit tests to prove this, but decided against it.